kotest-framework/kotest-framework-discovery/src/jvmMain/kotlin/io/kotest/framework/discovery/Discovery.kt

```diff
@@ -43,15 +42,12 @@ class Discovery(
 
    private val requests = ConcurrentHashMap<DiscoveryRequest, DiscoveryResult>()
 
-   // the results of a classpath scan, lazily executed and memoized.
-   private val scanResult = lazy { classgraph().scan() }
-
```

Comment:
This was never closed? Could be cause of the heap issues?

Reply:
Probably, but mostly in our own discovery tests. Classpath scanning was never part of known use cases in production (invocation via JUnit Platform by Gradle or Maven Surefire).